### PR TITLE
Responsive menu

### LINKED ISSUE #152: responsive menu (ux)
Menu layout does not work on narrow displays.

image

### PR DESCRIPTION
**Pull request by JulianWielga**
resolves #152

Zrzut ekranu 2020-01-21 o 16 17 51
Zrzut ekranu 2020-01-21 o 16 18 03
Zrzut ekranu 2020-01-21 o 16 34 03

### CODE REVIEW

File: ui/client/containers/EspApp.js [hunk 1]
```diff
@@ -133,7 +104,7 @@ class EspApp extends React.Component {
                         <Route path={Search.path} component={Search} exact/>
                         <Route path={Signals.path} component={Signals} exact/>
                         <Route path={AdminPage.path} component={AdminPage} exact/>
-                        <Route path={EspApp.path} component={Processes} exact/>
+                        <Redirect from={EspApp.path} to={Processes.path} exact/>
```
Comment (mproch) [inline ui/client/containers/EspApp.js]: Why this way?

Comment (JulianWielga) [reply ui/client/containers/EspApp.js]: To set URL to /processes. This way we get "Processes" active in menu

File: ui/client/containers/EspApp.js [hunk 2]
```diff
@@ -25,7 +27,7 @@ import "../stylesheets/main.styl"
 import "../app.styl"
 import ErrorHandler from "./ErrorHandler"
 
-class EspApp extends React.Component {
+export class EspApp extends React.Component {
```
Comment (lciolecki) [inline ui/client/containers/EspApp.js]: Please change name to NussknackerApp

File: ui/client/containers/EspApp.js [hunk 3]
```diff
@@ -70,52 +72,21 @@ class EspApp extends React.Component {
...
-              </ul>
-            </div>
-          </div>
-        </nav>
+        <div className="hide">{JSON.stringify(__GIT__)}</div>
+        <MenuBar
+            {...this.props}
+            app={EspApp}
+            leftElement={this.renderTopLeftButton()}
+            rightElement={this.environmentAlert(this.props.featuresSettings.environmentAlert)}
```
Comment (lciolecki) [inline ui/client/containers/EspApp.js]: Maybe extract featuresSettings from props? What if featuresSettings is null? I'm not sure that this field always has value.. :)

File: ui/client/containers/EspApp.js [hunk 4]
```diff
@@ -162,3 +133,4 @@ EspApp.path = `${nkPath}/`
 EspApp.header = "ESP"
```
Comment (lciolecki) [inline ui/client/containers/EspApp.js]: Please change ESP to Nussknacker :)

**Pull request merged**